### PR TITLE
Merge api classes

### LINKED ISSUE #26: Merge similar API classes
## ⚠️ Is your feature request related to a problem? Please describe
The API classes we use to communicate with the back-end are splittered and makes it more complicated in my opinion.

## 💡 Describe the solution you'd like
Make the classes cover multiple use cases, eg. one general to getting, adding and removing houses, one for house members, etc., things that are needed for all of the methods in the class should be in the constructor (eg. house id, token)

### PR DESCRIPTION
**Pull request by Tobi406**
## 📄 Motivation and Context
This merges some API classes together which I think makes them more easily readable.
I have also added some more components which (should hopefully) make it easier to build stuff.
Closes #26.

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jmprathab** [comment]
@Tobi406 Is it ready to be merged?

**Comment by Tobi406** [comment]
Pretty much, has some issues with member removal (see jmprathab/MyHome#56), but else it's ready.

File: src/pages/house/columns/detail-column.component.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          >
+
+          </Card>
+        ) : ''}
+      </>
+    )
+  }
+}
+
+export default DetailColumn;
```
Comment (Tobi406) [inline src/pages/house/columns/detail-column.component.jsx:39]:
```suggestion
export default DetailColumn;

```

File: src/api/Communities.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  getCommunities() {
+    console.trace(`Get communities api function called with token[${this.token}]`);
+    return axiosInstance.get(`/communities`, {
+      headers: {
+        'Authorization': `Bearer ${this.token}`,
+      },
+    });
+  }
+}
```
Comment (Tobi406) [inline src/api/Communities.js:26]:
```suggestion
}

```

File: src/api/CommunityAdmins.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  removeAdmin(adminId) {
+    console.trace(`Remove admins api function called with adminId[${adminId}], communityId[${this.communityId}], token[${this.token}]`);
+    return axiosInstance.delete(`/communities/${this.communityId}/admins/${adminId}`, {
+      headers: {
+        'Authorization': `Bearer ${this.token}`,
+      },
+    });
+  }
+}
```
Comment (Tobi406) [inline src/api/CommunityAdmins.js:38]:
```suggestion
}

```

File: src/api/Houses.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  getHousesOfCommunity(communityId) {
+    console.trace(`Get houses of community api function called with communityId[${communityId}], token[${this.token}]`);
+    return axiosInstance.get(`/communities/${communityId}/houses`, {
+      headers: {
+        'Authorization': `Bearer ${this.token}`,
+      },
+    });
+  }
+}
```
Comment (Tobi406) [inline src/api/Houses.js:58]:
```suggestion
}

```

File: src/api/Users.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  getUsers() {
+    console.trace(`Get users api function called with token[${this.token}]`);
+    return axiosInstance.get('/users', {
+      headers: {
+        'Authorization': `Bearer ${this.token}`,
+      }
+    });
+  }
+}
```
Comment (Tobi406) [inline src/api/Users.js:26]:
```suggestion
}

```

File: src/components/card/card.component.jsx [hunk 6]
```diff
@@ -3,21 +3,39 @@ import React, { Component } from "react";
...
+        ) : ''}
+        {this.props.children}
       </div>
-    );
+    )
   }
 }
 
-export default Card;
+export default Card;
```
Comment (Tobi406) [inline src/components/card/card.component.jsx:41]:
```suggestion
export default Card;

```

File: src/components/common/column.component.jsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  render() {
+    return (
+      <div className={`col-md-${this.props.cols}`}>
+        {this.props.children}
+      </div>
+    )
+  }
+}
+
+export default Column;
```
Comment (Tobi406) [inline src/components/common/column.component.jsx:13]:
```suggestion
export default Column;

```

File: src/components/common/content-between.component.jsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  render() {
+    return (
+      <div className="d-flex justify-content-between">
+        {this.props.children}
+      </div>
+    )
+  }
+}
+
+export default ContentBetween;
```
Comment (Tobi406) [inline src/components/common/content-between.component.jsx:13]:
```suggestion
export default ContentBetween;

```

File: src/components/common/page-row.component.jsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  render() {
+    return (
+      <Row className="mt-5">
+        {this.props.children}
+      </Row>
+    )
+  }
+}
+
+export default PageRow;
```
Comment (Tobi406) [inline src/components/common/page-row.component.jsx:14]:
```suggestion
export default PageRow;

```

**Pull request merged**